### DISCUSSION TITLE
getLatestOffset

### DISCUSSION THREAD

**Discussion by krasowskir** [spring-projects/spring-kafka#1663] [Ideas] [answered]
Hello, I stumbled over the problem, that KafkaTestUtils.getCurrentOffset() sometimes shows the offset=0 although there exists several records on a partition and I expect the offset to be 1. The offset value depends on the amount of commited messages and on the value auto.offset.reset. 

So I think it might be useful, if the class KafkaTestUtils would get a static method to show the latest offset like getCurrentOffset().

**Comment by artembilan** [comment]
>  although there exists several records on a partition

Looks like there is some misunderstanding what is `offset`. It is full not related to the number of records in the partition. The offset is a state of the consumer group. You may have several consumer groups on the same partition and each of it could stay on a different offset. That's why `getCurrentOffset()` API requires from us a `group` argument. 

What you are asking would sound more like a "number of records in the partition" to really align it properly with the `offset` concept. But I'm not sure if there is a way in Apache Kafka to obtain such a metric...

**Comment by garyrussell** [comment]
See [this stack overflow answer](https://stackoverflow.com/questions/65566470/calculate-delta-offsets-kafka-java/65567009#65567009) for how to get the end offsets.

We could add helper methods to the KTU.

**Comment by garyrussell** [comment, marked as answer]
https://github.com/spring-projects/spring-kafka/pull/1666
